enthought_example/example_mco/parameters.py

```diff
@@ -1,20 +1,43 @@
-from traits.api import Float
+from traits.api import Float, Range
+from traitsui.api import View, Item
 
-from force_bdss.api import (
-    BaseMCOParameter,
-    BaseMCOParameterFactory)
+from force_bdss.api import BaseMCOParameter, BaseMCOParameterFactory
 
 
 class RangedMCOParameter(BaseMCOParameter):
-    """Expresses a MCO parameter that has a range between two floating
-    point values."""
-    initial_value = Float(0.0)
-    lower_bound = Float(0.0)
-    upper_bound = Float(1.0)
+    """ Expresses a MCO parameter that has a range between two floating
+    point values.
+
+    """
+
+    lower_bound = Float()
+    upper_bound = Float()
+    initial_value = Range(
+        low="lower_bound",
+        high="upper_bound",
+        label="Initial value",
+        depends_on="lower_bound,upper_bound",
+    )
+
+    traits_view = View(Item("lower_bound"),
+                       Item("upper_bound"),
+                       Item("initial_value"))
+
+    def __init__(self, *args, **model_data):
```

Comment:
I guess with these ordering issues it wouldn't be unreasonable to just say: 
` __init__(self, low, high, *args, **model_data):`
`self.lower_bound = low etc.`
`super().__init__(*args, **model_data)`

Since this is a range it semms reasonable to require a lower and upper bound and then take things from there! My only reason for wanting to do this is that it might be easier to understand as example code (though honestly I'm not sure if it really is any easier!)

Reply:
Does this fit with the way that the force_bdss expects to create it? My understanding is that it takes all traits and passes them to the RangedMCOParameter's __init__ as **model_data, so I guess you either need the changes above, or you have to write a custom __init__ for the RangedMCOParameterFactory. Does that sounds right to you?